infoblox/resource_infoblox_ip_allocation.go

```diff
@@ -110,79 +109,60 @@ func resourceIPAllocation() *schema.Resource {
 					" used by Infoblox Terraform plugin to search for a NIOS's object" +
 					" which corresponds to the Terraform resource.",
 			},
+			"ref": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "NIOS object's reference, not to be set by a user.",
+			},
 		},
 	}
 }
 
-func getAndRenewHostRecAltId(d *schema.ResourceData, m interface{}, updateId bool) (hostRec *ibclient.HostRecord, err error) {
+// This function is for retrieving a host record by either known reference or,
+// if the reference points to nothing (returns 'not found'),
+// by internal_id. It returns the host record itself.
+//
+// If err == nil then hostRec != nil,
+// other cases must be considered as a serious bug.
+// err == nil also means that 'internal_id' exists and
+// is of a proper format.
+// This function MUST NOT set any resource's properties,
+// other behaviour is a bug.
+func getOrFindHostRec(d *schema.ResourceData, m interface{}) (
```

Comment:
Maybe we should pass `d` as a `schema.ResourceData` instead of a pointer to the same struct. I think that this way, function will not set resource properties by-design.

```suggestion
func getOrFindHostRec(d schema.ResourceData, m interface{}) (
```